### PR TITLE
Pimp the docs

### PR DESCRIPTION
**Pull request by TimothyGillespie**
I haven't proof-read it yet, but even with errors it's still good to have.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,35 +4,90 @@ NodeJS >=18.19.0
...
 ```
 
 ## Local Development
 
-1. **Start the Development Server:** Use the command `npm run dev` to launch `npm run dev:frontend` and `netlify-cms` concurrently.
+**Start the Development Server:** Use the command `npm run dev` to launch `npm run dev:frontend` and `netlify-cms` concurrently.
+
+### Previewing the Website
+
+You can now navigate to [http://localhost:3000](http://localhost:3000) to see a local preview of the website including your changes. The port may change when 3000 is taken - to check the console for any hint of that.
```
Comment (Pixelguymm) [inline README.md:19]: These don't need to be hyperlinks.

File: README.md [hunk 2]
```diff
@@ -4,35 +4,90 @@ NodeJS >=18.19.0
...
+│   ├── characteristics // Folder containing the images for the diffs on the map cards
+│   ├── favicon.png // This file is the favicon of the website
+│   ├── uploads // Directory, that contains uploaded images from netlify CMS
+├── svelte.config.js // Svelte(kit) configuration - contains the configurations for crawling and SSG as well
+├── update-cover-url-overwrites.sh // See "Utility Scripts" for info
+```
+
+## Project Routing
+
+The routes are determined by file path starting at `src/routes`. For example, (`src/routes/community-hub/+page.svelte`)[src/routes/community-hub/+page.svelte] renders under (https://bsaber.com/community-hub)[https://bsaber.com/community-hub] or in the local development environment under (http://localhost:3000/community-hub)[http://localhost:3000/community-hub].
```
Comment (Pixelguymm) [inline README.md:63]: This first hyperlink has its brackets flipped, the format for hyperlinks is `[text](url)`.
The second and third, like the ones above, don't need to be hyperlinks explicitly.

File: README.md [hunk 3]
```diff
@@ -4,35 +4,90 @@ NodeJS >=18.19.0
...
-### Notes
-- The `npm run dev:frontend` command runs your application and integrates netlify under the `/admin/index.html` routes. 
+- The `npm run dev:frontend` command runs your application and integrates netlify under the `/admin/index.html` routes.
 - The CMS proxy server enables the use of the local strategy for content management, directly saving files to the local file system.
 
+## Short File Tree Description
+
+```txt
+.
+├── environmentVariables.ts.template // Template file to inject .env variables. Will be filed wit envsubst in the build step. See "Building Details" for more info.
```
Comment (Pixelguymm) [inline README.md:36]: "filed wit" → "filled with"

File: README.md [hunk 4]
```diff
@@ -4,35 +4,90 @@ NodeJS >=18.19.0
...
+├── migration-from-bsaber.com // Scripts used to transfer MotW from the old page - can be ignored
+├── README.md
+├── src
+│   ├── app.html // Entry point for the application - contains placeholder for special tags such as <svelte:head> as well
+│   ├── app.scss // Global styles
+│   ├── collections // Actual data for the collection per their model definiton. One folder = One Collection
```
Comment (Pixelguymm) [inline README.md:42]: Some weird capitalization at the end there.

File: README.md [hunk 5]
```diff
@@ -4,35 +4,90 @@ NodeJS >=18.19.0
...
+│   ├── lib // Components and functions that can be accessed with the $lib path
+│   ├── maps.ts // Central point for enums, that are shared across multiple components
```
Comment (Pixelguymm) [inline README.md:44]: Unless "enums" is short for "enumerables" in this case, I don't think this is correct. All objects in that file are maps, or `Record`s in TypeScript terms.

File: README.md [hunk 6]
```diff
@@ -4,35 +4,90 @@ NodeJS >=18.19.0
...
+│   ├── routes // Svelte(kit) routing - these are the pages of our website. See the "Project Routing" section for more details
```
Comment (Pixelguymm) [inline README.md:45]: Missing punctuation.

File: README.md [hunk 7]
```diff
@@ -4,35 +4,90 @@ NodeJS >=18.19.0
...
+│   │   ├── 404.svelte
+│   ├── scss // Styles that are shared across multiple components
+│   │   └── variables.scss // Defines a preset of colors and values to be used uniformly across the project - need a color? Import from here!
+│   └── types.ts // A central place for types - if types are specific for a component (or similar) and not shared across the project, put them in the respective component file
+├── static // These files will be mounted on the root. For example, ./static/example.jpg will later be accessible under https://bsaber.com/example.jpg
+│   ├── admin
+│   │   ├── config.yml // This files defines the config and data model for Netlify CMS - if you want to edit, add or remove a collection some changes will happen here
```
Comment (Pixelguymm) [inline README.md:52]: "This files" → "This file"

File: README.md [hunk 8]
```diff
@@ -4,35 +4,90 @@ NodeJS >=18.19.0
...
+│   │   └── index.html // Entry point for the Netlify CMS page for admins
+│   ├── characteristics // Folder containing the images for the diffs on the map cards
+│   ├── favicon.png // This file is the favicon of the website
+│   ├── uploads // Directory, that contains uploaded images from netlify CMS
```
Comment (Pixelguymm) [inline README.md:56]: Unnecessary comma.

File: README.md [hunk 9]
```diff
@@ -4,35 +4,90 @@ NodeJS >=18.19.0
...
+├── svelte.config.js // Svelte(kit) configuration - contains the configurations for crawling and SSG as well
```
Comment (Pixelguymm) [inline README.md:57]: Did you mean "SSR" or is "SSG" a term I don't know about?

File: README.md [hunk 10]
```diff
@@ -4,35 +4,90 @@ NodeJS >=18.19.0
...
+├── update-cover-url-overwrites.sh // See "Utility Scripts" for info
+```
+
+## Project Routing
+
+The routes are determined by file path starting at `src/routes`. For example, (`src/routes/community-hub/+page.svelte`)[src/routes/community-hub/+page.svelte] renders under (https://bsaber.com/community-hub)[https://bsaber.com/community-hub] or in the local development environment under (http://localhost:3000/community-hub)[http://localhost:3000/community-hub].
+
+Note the following when it comes to those:
+
+1. **We Statically Generate our Site**: You cannot rely on server side routing to dynamically update the pages. These only execute once when rendering and generating the static HTML and CSS files and cannot react to, for example, a change in the file system. Creating and managing files will thus also not usually make sense.
```
Comment (Pixelguymm) [inline README.md:67]: I don't think these need to be in title case, and only the first three are anyway.

File: README.md [hunk 11]
```diff
@@ -4,35 +4,90 @@ NodeJS >=18.19.0
...
+
+2. **Pages with Path Parameters Need to be Linked**: Paths that contain path parameters such as `[page]` need to be linked to somehwere from another page that either has no path parameter or is linked by a page fulfilling these conditions. Basically, they need to be crawlable by following links. If a page cannot be crawled starting from any non-parameter path it will simply exist in the final product.
```
Comment (Pixelguymm) [inline README.md:69]: "somehwere" → "somewhere"

File: README.md [hunk 12]
```diff
@@ -4,35 +4,90 @@ NodeJS >=18.19.0
...
+
+3. **`npm run dev` =/= Production Environment**: You cannot rely on it 100%. For example, non-rendered pages will still work with `npm run dev` since it actually executes server-side.
```
Comment (Pixelguymm) [inline README.md:71]: Just a suggestion, you could use `≠` here.

File: README.md [hunk 13]
```diff
@@ -4,35 +4,90 @@ NodeJS >=18.19.0
...
+
+4. **You usually want to stick to `+page.server.ts` not `+page.ts`**: The difference between the two is, that the `.server.ts` only executes server side (so once when rendering). The `+page.ts` will run server and client side. Due to the way we use server-side code it usually doesn't make sense to use `+page.ts` and can quickly result in confusing errors.
```
Comment (Pixelguymm) [inline README.md:73]: The comma should be placed before "is".

File: README.md [hunk 14]
```diff
@@ -4,35 +4,90 @@ NodeJS >=18.19.0
...
+
+5. **API functionality won't get you far**: Svelte allows you to define an API on the server side. This will also work only very limitedly. At the very list the dynamic nature of it will be stripped away.
```
Comment (Pixelguymm) [inline README.md:75]: "list" → "least"

File: README.md [hunk 15]
```diff
@@ -4,35 +4,90 @@ NodeJS >=18.19.0
...
 
 ## Contributing
 
-### Before submitting a PR
+We are available for question, support and discussion on our (Discord server)[https://discord.gg/VJZHUbt].
```
Comment (Pixelguymm) [inline README.md:79]: This hyperlink and the next are flipped as well.

File: README.md [hunk 16]
```diff
@@ -4,35 +4,90 @@ NodeJS >=18.19.0
...
 
-Verify that the app can be built:
+We would hate for you to spend valuable time on something that we end up not using. Here are some infos on how to increase the likelihood of us merging your changes:
```
Comment (Pixelguymm) [inline README.md:81]: "are some infos" → "is some info", "information" cannot be plural.

File: README.md [hunk 17]
```diff
@@ -4,35 +4,90 @@ NodeJS >=18.19.0
...
 
-```
-npm run build
-```
+- You can see the tasks that we decided to work and which are not worked on by anyone else as (approved, but non-assigned Issues)[https://github.com/beastsaber/bsaber/issues?q=is%3Aopen+is%3Aissue+label%3Aapproved+no%3Aassignee]. Do watch the labels!
 
-Run prettier formatter:
+- If you are unsure if you can work on a task or what it entails: Do contact us. Either comment on the issue or drop by on the Discord. We are for most questions and any doubt on your part!
```
Comment (Pixelguymm) [inline README.md:85]: The last sentence is missing a word, could be: "We are happy to answer your questions and concerns."

File: README.md [hunk 18]
```diff
@@ -4,35 +4,90 @@ NodeJS >=18.19.0
...
 
-```
-npm run format
-```
+- Make us aware, that you'd like to work on the issue. We'll assign it to you and avoid the work being done twice!
```
Comment (Pixelguymm) [inline README.md:87]: Unnecessary comma.

**Review by Pixelguymm — CHANGES_REQUESTED** [review]

File: README.md [hunk 19]
```diff
@@ -4,35 +4,90 @@ NodeJS >=18.19.0
...
-### Before submitting a PR
+We are available for question, support and discussion on our (Discord server)[https://discord.gg/VJZHUbt].
 
-Verify that the app can be built:
+We would hate for you to spend valuable time on something that we end up not using. Here are some infos on how to increase the likelihood of us merging your changes:
 
-```
-npm run build
-```
+- You can see the tasks that we decided to work and which are not worked on by anyone else as (approved, but non-assigned Issues)[https://github.com/beastsaber/bsaber/issues?q=is%3Aopen+is%3Aissue+label%3Aapproved+no%3Aassignee]. Do watch the labels!
```
[the remark this answers is not in the document]
Comment (TimothyGillespie) [reply README.md:83]: True, I like this better.

**Comment by TimothyGillespie** [comment]
You're simply correct. This was very raw. Thanks for the thorough review! I have updated them.

File: README.md (hunk 13 above)
Comment (TheCzar1994) [reply README.md:73]: I would say there should be no comma here at all

**Review by TheCzar1994 — CHANGES_REQUESTED** [review]
I've re-reviewed this from Pixel's review. Most of the changes have been made but there are some requested changes that are still outstanding, mostly just grammatical corrections. Can be pushed once those are finished up though.

File: README.md (hunk 9 above)
Comment (TimothyGillespie) [reply README.md:57]: It's for Statical Site Generation. It's intended. We only do SSR when generating the static site. 
Proof: https://kit.svelte.dev/docs/adapter-static

**Review by TheCzar1994 — APPROVED** [review]

**Pull request merged**